Mathlib/Order/Interval/Finset/Basic.lean

```diff
@@ -446,6 +471,19 @@ theorem Icc_eq_singleton_iff : Icc a b = {c} ↔ a = c ∧ b = c := by
 theorem Ico_disjoint_Ico_consecutive (a b c : α) : Disjoint (Ico a b) (Ico b c) :=
   disjoint_left.2 fun _ hab hbc => (mem_Ico.mp hab).2.not_le (mem_Ico.mp hbc).1
 
+@[simp]
+theorem Ici_top [OrderTop α] : Ici (⊤ : α) = {⊤} :=
+  Icc_eq_singleton_iff.2 ⟨rfl, rfl⟩
+
+@[simp]
+theorem Iic_bot [OrderBot α] : Iic (⊥ : α) = {⊥} :=
+  Icc_eq_singleton_iff.2 ⟨rfl, rfl⟩
+
+@[simp]
+theorem Icc_bot_top [BoundedOrder α] [Fintype α] : Icc (⊥ : α) ⊤ = univ := by
```

Comment:
Can you add separate `Icc_bot`/`Icc_top`/`Ico_bot`/`Ioc_top` lemmas that rewrite to a single-sided interval? I think we already have those for `Set`